src/test/regress/expected/foreign_tables_mx.out

```diff
@@ -416,6 +416,62 @@ NOTICE:  renaming the new table to foreign_tables_schema_mx.foreign_table_local_
 (1 row)
 
 DROP FOREIGN TABLE foreign_table_local;
+-- disallow dropping table_name when foreign table is in metadata
+CREATE TABLE table_name_drop(id int);
+CREATE FOREIGN TABLE foreign_table_name_drop_fails (
+        id INT
+)
+        SERVER foreign_server_local
+        OPTIONS (schema_name 'foreign_tables_schema_mx', table_name 'table_name_drop');
+SELECT citus_add_local_table_to_metadata('foreign_table_name_drop_fails');
+ citus_add_local_table_to_metadata
+---------------------------------------------------------------------
+
+(1 row)
+
+-- table_name option is already added
+ALTER FOREIGN TABLE foreign_table_name_drop_fails OPTIONS (ADD table_name 'table_name_drop');
+ERROR:  option "table_name" provided more than once
+-- throw error if user tries to drop table_name option from a foreign table inside metadata
+ALTER FOREIGN TABLE foreign_table_name_drop_fails OPTIONS (DROP table_name);
+ERROR:  alter foreign table alter options (drop table_name) command is not allowed
+-- other options are allowed to drop
+ALTER FOREIGN TABLE foreign_table_name_drop_fails OPTIONS (DROP schema_name);
+CREATE FOREIGN TABLE foreign_table_name_drop (
+        id INT
+)
+        SERVER foreign_server_local
+        OPTIONS (schema_name 'foreign_tables_schema_mx', table_name 'table_name_drop');
+-- user can drop table_option if foreign table is not in metadata
+ALTER FOREIGN TABLE foreign_table_name_drop OPTIONS (DROP table_name);
+-- we should not intercept data wrappers other than postgres_fdw
+CREATE EXTENSION file_fdw;
+-- remove validator method to add table_name option; otherwise, table_name option is not allowed
+SELECT result FROM run_command_on_all_nodes('ALTER FOREIGN DATA WRAPPER file_fdw NO VALIDATOR');
+           result
+---------------------------------------------------------------------
+ ALTER FOREIGN DATA WRAPPER
+ ALTER FOREIGN DATA WRAPPER
+ ALTER FOREIGN DATA WRAPPER
+(3 rows)
+
+CREATE SERVER citustest FOREIGN DATA WRAPPER file_fdw;
+\copy (select i from generate_series(0,100)i) to '/tmp/test_file_fdw.data';
+CREATE FOREIGN TABLE citustest_filefdw (
+        data text
+)
+        SERVER citustest
+        OPTIONS ( filename '/tmp/test_file_fdw.data');
+-- add non-postgres_fdw table into metadata even if it does not have table_name option
+SELECT citus_add_local_table_to_metadata('citustest_filefdw');
+ citus_add_local_table_to_metadata
+---------------------------------------------------------------------
+
+(1 row)
+
+ALTER FOREIGN TABLE citustest_filefdw OPTIONS (ADD table_name 'unused_table_name_option');
+-- drop table_name option of non-postgres_fdw table even if it is inside metadata
+ALTER FOREIGN TABLE citustest_filefdw OPTIONS (DROP table_name);
```

Comment:
let's clean up what we create (the server is not created in the schema?)

Reply:
It is created in the schema.